src/System.Net.Http/src/System/Net/Http/HttpContent.cs

```diff
@@ -315,8 +310,12 @@ public Task CopyToAsync(Stream stream, TransportContext context)
                 Task task = null;
                 if (IsBuffered)
                 {
-                    byte[] data = this.GetDataBuffer(_bufferedContent);
-                    task = stream.WriteAsync(data, 0, (int)_bufferedContent.Length);
+                    task = _bufferedContent.CopyToAsync(stream).ContinueWithStandard(_bufferedContent, (copyTask, state) =>
```

Comment:
Also, in the previous version if the stream.WriteAsync failed, ```task``` would be a faulted task that contained the error.  With this new version, ```task``` is always going to be a successfully completed task even if the underlying ```WriteAsync``` fails.